### DISCUSSION TITLE
Keybind / method set width to "fill rest of screen"?

### DISCUSSION THREAD

**Discussion by raiguard** [YaLTeR/niri#599] [Q&A] [answered]
I have a fairly unique requirement: I am streaming Factorio gameplay using niri, which requires me to make Factorio exactly 1920x1080. This part I have solved already, but what I would like to set obs' width to fill the rest of the screen space.

I understand that this kind of thing is usually better served by i3/sway, but I can't use sway because it doesn't support pipewire window capture. Is there a way I can accomplish this?

I would imagine something like `fill` in addition to the existing `proportion` and `fixed` layout rules.

![image](https://github.com/user-attachments/assets/741774e3-c762-47c7-8589-2258210ac302)

**Comment by YaLTeR** [comment]
I thought about this. The question here is, what happens when there's more windows on the workspace. E.g. in this case, do you want it to fill only to the right of the screen, or also fill so that the left window disappears?

![image](https://github.com/user-attachments/assets/9befd9d8-353d-4dd3-ae5e-6dcaacb5106c)

What if you want to fill in such a way that there are 3 windows on screen? What does the key bind action parameters look like?

**Reply by raiguard** [reply]
When you invoke the keybind, the currently focused window would stretch to fill the rest of the current gap on the screen at the time of invocation. So in the pictured case, your terminal would stretch to the right to fill the screen.

In this case, firefox would stretch to fill the gap:
![image](https://github.com/user-attachments/assets/183b4dc6-c638-4895-b51f-791891d1f8fb)

**Reply by YaLTeR** [reply]
So this bind will only work when there's any free space on the screen, and never in the middle of a layout?

**Reply by raiguard** [reply]
Now that I look closer at your original example I see what you were getting at.

![image](https://github.com/user-attachments/assets/c54e6e94-78f6-4164-b86b-e2a908670896)

In this case I would expect "fill" to stretch Firefox both to the right and left so that there are no partially-visible windows remaining on-screen.

![image](https://github.com/user-attachments/assets/a09497f2-fa5a-4e70-aab8-46bba1553ba7)

In this case it would do the same thing - stretch to the left to fill that partially visible window space, and stretch right to fill that empty space.

**Reply by YaLTeR** [reply]
So the way it should work is to stretch to fill out any area covered by either nothing or partially visible windows, and leave all fully visible windows intact?

**Reply by raiguard** [reply]
Correct.

**Reply by YaLTeR** [reply]
Should be doable.

**Reply by YaLTeR** [reply]
Any naming suggestions for the bind? :)

**Reply by raiguard** [reply]
Could either be `set-column-width fill` or `fill-column-width`. I definitely think `fill` fits in the name.

**Reply by YaLTeR** [reply]
Opened an issue: https://github.com/YaLTeR/niri/issues/976

**Comment by YaLTeR** [comment, marked as answer]
Implemented in 92cc2b89f75fd13d83208626fa49724216fbbf9e. Please give it a try, see if it works as you expect.

**Reply by raiguard** [reply]
Seems to work great! Thank you!

Unfortunately, I went back to Sway because its workflow suits me better. I really find myself missing the gnome integration and animations of Niri though. :sob: 

Keep up the great work!

**Reply by YaLTeR** [reply]
I see, thanks for testing!
